src/iconv.cr

```diff
@@ -13,15 +13,19 @@ struct Iconv
       to = "#{to}//IGNORE"
     end
 
-    Errno.value = 0
     @iconv = LibC.iconv_open(to, from)
-    if Errno.value != 0
-      if original_from == "UTF-8"
-        raise ArgumentError.new("invalid encoding: #{original_to}")
-      elsif original_to == "UTF-8"
-        raise ArgumentError.new("invalid encoding: #{original_from}")
+
+    if @iconv.address == LibC::SizeT.new(-1)
```

Comment:
Maybe `Pointer#address`'s type should change depending on the platform, I'm not sure.